catalog/app/containers/Admin/UsersAndRoles/SsoConfig.tsx

```diff
@@ -0,0 +1,171 @@
+import * as FF from 'final-form'
+import * as React from 'react'
+import * as RF from 'react-final-form'
+import * as M from '@material-ui/core'
+import * as Lab from '@material-ui/lab'
+
+import Lock from 'components/Lock'
+import { loadMode } from 'components/FileEditor/loader'
+import type * as Model from 'model'
+import type * as Dialogs from 'utils/GlobalDialogs'
+import * as GQL from 'utils/GraphQL'
+import assertNever from 'utils/assertNever'
+import * as validators from 'utils/validators'
+
+import SET_SSO_CONFIG_MUTATION from './gql/SetSsoConfig.generated'
+import SSO_CONFIG_QUERY from './gql/SsoConfig.generated'
+
+const TextEditor = React.lazy(() => import('components/FileEditor/TextEditor'))
+
+type TextFieldProps = RF.FieldRenderProps<string> & M.TextFieldProps
+
+const TEXT_EDITOR_TYPE = { brace: 'yaml' as const }
+
+const ERRORS = {
+  required: 'Enter an SSO config',
+}
+
+function TextField({ errors, input, meta }: TextFieldProps) {
+  // TODO: lint yaml
+  const errorMessage = meta.submitFailed && errors[meta.error]
```

Comment:
does this handle submit errors properly? i vaguely remember i had some confusion with form errors vs submit errors -- they were under different fields and i had to check both these fields, but i don't remember which api this was, maybe form-level 🤷

Reply:
This is for field-level errors. And I just don't show the validation error when form hadn't submitted.
This piece of code is the same as in other TextField across app.

On the other hand, form-specific error. There I made a "shortcut" and since this form is one-field, I show the error from API directly, not from final-form.